### DISCUSSION TITLE
Is it possible to set up SSL with a custom domain on github pages

### DISCUSSION THREAD

**Discussion by AlMcKinlay** [community/community#21897] [Other Feature Feedback, Questions, & Ideas]
<p>When setting up a GitHub pages, I tried to enable SSL, but because I have a custom domain, it doesn’t let me do this. Is there a way we can set this up?</p>

**Comment by osioke** [comment, marked as answer]
<p>Yes, sure you can.</p>
<p>I wanted to do this a while back and did some research on it. It appears you can use CloudFlare to do this, by setting a page rule that forces HTTPS redirects amongst other things.</p>
<p>There are quite a lot of tutorials out there, but <a href="https://blog.cloudflare.com/secure-and-fast-github-pages-with-cloudflare/" title="Secure and fast GitHub Pages with CloudFlare" rel="nofollow noopener">this tutorial</a> on CloudFlare’s blog as well as this <a href="https://gist.github.com/cvan/8630f847f579f90e0c014dc5199c337b" title="How to serve a custom HTTPS domain on GitHub Pages with CloudFlare" rel="nofollow noopener">GitHub Gist</a> really broke it down.</p>
<p>Do let me know if it works well when you try it.</p>
<p>Cheers!</p>

**Comment by AlMcKinlay** [comment]
<p>Great, thanks. I’ll give that a try, that makes sense.</p>

**Comment by lucypero** [comment]
<p>You can do it with <a href="https://fly.io/" rel="nofollow noopener">fly.io</a> too.</p>

**Comment by osioke** [comment]
<p>Cool, I haven’t heard of them before, I will check them out.</p>

**Comment by gionn** [comment]
<p>Tried <a href="http://fly.io" rel="nofollow noopener">fly.io</a>, but it looks like they don’t support letsencrypt directly.</p>
